public/app/features/templating/variable_srv.ts

```diff
@@ -303,6 +311,42 @@ export class VariableSrv {
     this.$location.search(params);
   }
 
+  storeUpdated() {
+    const state = store.getState();
+
+    let path = this.originalUrl;
+    if (path.indexOf('?') >= 0) {
+      path = path.slice(0, this.originalUrl.indexOf('?'));
+    }
+    if (state.location.path === path) {
+      if (state.location.url === this.originalUrl) {
+        return this.$q.when({});
+      }
+
+      this.originalUrl = this.$location.url();
+      const queryParams = this.$location.search();
```

Comment:
When updating via `getLocationSrv().update( { ... } )` this works when I change the above line to:
```
const queryParams = state.location.query; //this.$location.search();
```

@mtanda -- for your test cases does that work?  or does it need to be $location.search()?